src/new/wallet.js

```diff
@@ -2565,6 +2565,91 @@ class HathorWallet extends EventEmitter {
 
     return graphvizData;
   }
+
+  /**
+   * This function is responsible for getting the details of each token in the transaction.
+   * @param {Object} wallet the current wallet
+   * @param {{ [tokenUid]: number }} tokenBalances Object with the token uid as key and the balance as value
+   * @returns {Promise<Array<{ txId: string, timestamp: number, version: number, voided: boolean, weight: number, tokenName: string, tokenSymbol: string, balance: number }>> } Array of token details
+   * @example
+   * [
+   *  {
+   *   txId: '000021e7addbb94a8e43d7f1237d556d47efc4d34800c5923ed3a75bf5a2886e';
+   *   timestamp: 123456789;
+   *   version: 1;
+   *   voided: false;
+   *   weight: 18.5;
+   *   tokenId: '00',
+   *   tokenName: 'Hathor',
+   *   tokenSymbol: 'HTR',
+   *   balance: 100,
+   *  },
+   * ]
+   */
+  async getTxById(txId) {
+    /**
+     * This function is responsible for getting the details of each token in the transaction.
+     * @param {Object} wallet the current wallet
+     * @param {{ [tokenUid]: number }} tokenBalances Object with the token uid as key and the balance as value
+     * @returns {Promise<Array<{ txId: string, timestamp: number, version: number, voided: boolean, height: number, weight: number, tokenName: string, tokenSymbol: string, balance: number }>> } Array of token details
+     * @example
+     * [
+     *  {
+     *   txId: '000021e7addbb94a8e43d7f1237d556d47efc4d34800c5923ed3a75bf5a2886e',
+     *   timestamp: 1673039453,
+     *   version: 1;
+     *   voided: false;
+     *   weight: 18.5;
+     *   tokenId: '00',
+     *   tokenName: 'Hathor',
+     *   tokenSymbol: 'HTR',
+     *   balance: 100,
+     *  },
+     * ]
+     */
+    const getTokenDetails = async (wallet, tx, tokenBalances) => {
+      const tokenUids = Object.keys(tokenBalances);
+      try {
+        const tokenDetails = await Promise.all(
+          tokenUids.map(async (token) => {
```

Comment:
This will attempt to download all `tokenDetails` in parallel, right? What happens if there are 255 different tokens? Should we limit the number of parallel downloads to something like, 5?

Not sure if this is a problem though

Reply:
The transactions have at most 16 tokens but this is still a problem in my opinion.

I have a suggestion to improve this method but I feel we could talk first to understand one situation

Reply:
Removed.